src/modules/rest/extjs/src/test/java/it/geosolutions/geostore/services/rest/impl/RESTExtJsServiceImplTest.java

```diff
@@ -362,14 +376,431 @@ public void testGetAllResources_unadvertised() throws Exception {
         }
     }
 
+    @Test
```

Comment:
Great coverage for new search/filter/sort features, verifying the new code paths (sorting by date, filtering by group, date-based filters, etc.).

Overall, these tests do a good job hitting the main functionality. Just confirm you also test negative scenarios or edge conditions that might raise exceptions (BadRequestServiceEx, IllegalStateException, etc.).

Suggestions for Additional Test Cases:

- Empty Results: Check what happens if no resource matches the filter.
- Invalid sortBy: If the user tries sorting by a non-existing field.
- Null filter or filter with null names in GroupFilter.
- Security Edge Cases: A resource that has no security rules, or a user with no groups, etc.

Reply:
> Empty Results: Check what happens if no resource matches the filter.

Added test cases to check empty result in this case.

> Invalid sortBy: If the user tries sorting by a non-existing field.

The exception thrown here is already handled and silenced by the REST service method, returning `null`. Added test anyway.

> Null filter or filter with null names in GroupFilter.

Added specific tests, see [here](https://github.com/geosolutions-it/geostore/pull/392#discussion_r1920268922).

> Security Edge Cases: A resource that has no security rules, or a user with no groups, etc.

Added specific tests for other corner cases.